wg-naming/README.md

```diff
@@ -0,0 +1,70 @@
+<!---
+This is an autogenerated file!
+
+Please do not edit this file directly, but instead make changes to the
+sigs.yaml file in the project root.
+
+To understand how this file is generated, see https://git.k8s.io/community/generator/README.md
+--->
+# Naming Working Group
+
+
+## Stakeholder SIGs
+* SIG Architecture
+* SIG Contributor Experience
+* SIG Docs
+
+
+
+## Organizers
+
+* Celeste Horgan (**[@celestehorgan](https://github.com/celestehorgan)**), CNCF
+* Stephen Augustus (**[@justaugustus](https://github.com/justaugustus)**), VMware
+* Zach Corleissen (**[@zacharysarah](https://github.com/zacharysarah)**), Linux Foundation
+
+## Contact
+- Slack: [#wg-naming](https://kubernetes.slack.com/messages/wg-naming)
+- [Mailing list](https://groups.google.com/forum/#!forum/kubernetes-wg-naming)
+- [Open Community Issues/PRs](https://github.com/kubernetes/community/labels/wg%2Fnaming)
+<!-- BEGIN CUSTOM CONTENT -->
+
+**The following section will be reworked and formalized as a charter once the
+Working Group has been approved by the Steering Committee.**
+
+### Goals
+
+- Evaluate language and naming choices within the Kubernetes organization, with
+  a specific initial focus of:
+  - Removing language which cannot be fully separated from racist, sexist or
+    homophobic connotations in cultural contexts outside of computer
+    programming
+  - Improving clarity of codebases and documentation by replacing idioms,
+    metaphors and slang specific to the English language
+- Create a list of terms to avoid and propose alternatives
+- Define a process by which any member of the Kubernetes organization can
+  propose a language recommendation, have them evaluated based on a rubric, and
+  see them implemented across all codebases.
+  - Provide an easily findable location for language recommendations and
+    follow-up issues, similar to an architectural decision record.
+- Work with stakeholder SIGs to implement the changes recommended. We
+  anticipate the following:
+  - Provide stakeholder SIGs with guidance on naming, language conventions, and
+    processes
+  - Collaborate with SIG Architecture and other stakeholders on an
+    implementation timeline and strategies for dealing with follow-up issues
+    from renaming, like deprecations
+  - Collaborate with SIG Docs and SIG Contributor Experience on documenting
+    language recommendations and processes
+  - Work with the Code of Conduct Committee to add code architecture to the COC
+
+### Dissolution Criteria
+
+When this WG dissolves, the Kubernetes community should have:
+
+- A process in place to evaluate language changes on an ongoing basis
+- A list of terms to avoid in codebases across the organization
+- A timeline on which to replace component names in the kubernetes/kubernetes
```

Comment:
It seems like sustainability after this working group disbands would require that the process create above and the future owners of that process, would need to be able to handle implementation and timeline.  Could we use that process and those owners to drive the first set of changes as well?  If not, can we have confidence that this effort will be self-sustaining over time?

See the comment from Celeste in this PR about avoiding the need to reform this group for the next cultural moment: https://github.com/kubernetes/community/pull/4884/files#r445043710

Reply:
What does "on which to replace" mean for such a timeline in an OpenSource project? Who drives it? Who contributes the resources to implement changes? I could imagine that the wg will output a plan with parties committed to implement it. I guess that's what this means, and not an order to be finished by date X.

Reply:
Working with the stakeholders and code owners will be key. Yes, this effort will need to be resourced. We've had lots of interest and volunteers to help with this, but ensuring we do this in a way that doesn't introduce bugs/regressions will need effort from code owners.

What that timeline is and how it will be resourced is an open question for the working group (and one they can answer after formation IMO)

Reply:
I offer to program-manage this effort to the extent of bringing the stakeholders and parties together to discuss a strategy for getting the work moving, tracking the work (including visualisations and self-reporting documents), and updating on progress. 

One of my first steps would be to bring folks together to discuss implementation strategy — how and where does the work start from a technical perspective.

Reply:
(Added as a post-formation discussion point.)